be/src/agent/agent_server.cpp

```diff
@@ -43,9 +46,15 @@ const uint32_t REPORT_TASK_WORKER_COUNT = 1;
 const uint32_t REPORT_DISK_STATE_WORKER_COUNT = 1;
 const uint32_t REPORT_OLAP_TABLE_WORKER_COUNT = 1;
 const uint32_t REPORT_WORKGROUP_WORKER_COUNT = 1;
+const uint32_t TASK_FINISH_MAX_RETRY = 3;
+const uint32_t ALTER_FINISH_TASK_MAX_RETRY = 10;
+
+FrontendServiceClientCache g_master_service_client_cache;
 
 AgentServer::AgentServer(ExecEnv* exec_env, const TMasterInfo& master_info)
-        : _exec_env(exec_env), _master_info(master_info) {
+        : _exec_env(exec_env),
+          _master_info(master_info),
+          _master_client(new MasterServerClient(master_info, &g_master_service_client_cache)) {
```

Comment:
I have a question, when the master_info changes, the _master_client will send request to the old master which is not correct. I think it should be that generate the right client with the right master_info when sending request rather than using an old master_client pointing to the old master.

Reply:
`_master_client` is holding a reference to the `master_info` and the `master_info` will be updated by the HeartbeatServer.